nuto/mechanics/tools/QuasistaticSolver.cpp

```diff
@@ -26,16 +30,16 @@ void QuasistaticSolver::SetConstraints(Constraint::Constraints constraints)
 {
     mConstraints = constraints;
     if (mX[mDofs.front()].rows() == 0)
-        mX = mProblem.RenumberDofs(constraints, mDofs, GlobalDofVector()).J;
+        mX = mProblem.RenumberDofs(constraints, mDofs, DofVector<double>());
     else
-        mX = mProblem.RenumberDofs(constraints, mDofs, ToGlobalDofVector(ToEigen(mX, mDofs))).J;
+        mX = mProblem.RenumberDofs(constraints, mDofs, mX);
 
     for (auto dofI : mDofs)
         for (auto dofJ : mDofs)
             if (dofI.Id() == dofJ.Id())
-                mCmat(dofI, dofI) = constraints.BuildConstraintMatrix(dofI, mX[dofI].rows());
+                mCmatUnit(dofI, dofI) = constraints.BuildUnitConstraintMatrix2(dofI, mX[dofI].rows());
```

Comment:
As far as I can tell, `mCmatUnit` is not used anymore. It is rebuild on every solve. I prefer keeping `mCmatUnit` here and adding the possibility to pass an existing `Cmat` to `Solve`.